core/contracts/test/Shield.test.js

```diff
@@ -0,0 +1,76 @@
+import { getAccounts, getSigner, getProvider } from './utils';
+import { ethers, utils } from 'ethers';
+
+import ShieldArtifact from '../artifacts/Shield.json';
+import VerifierArtifact from '../artifacts/VerifierNoop.json';
+
+const treeHeight = 4;
+let verifier, shield;
+let accounts, signer;
+let root;
+
+test('Deploy Verifier contract', async () => {
```

Comment:
needs to test that insertLeaf is not available without proof

Reply:
Added test